### PR TITLE
(CB2-12581): Update SNS topic retro-gen to use raw message delivery

### PR DESCRIPTION
**Pull request by Daniel-Searle**
## Update SNS topic retro-gen to use raw message delivery

This change is to amend the logic for passing of the event event to cater for the raw delivery change.

Related issue: [CB2-12581](https://dvsa.atlassian.net/browse/CB2-12581)

## Checklist

- [ ] Branch is rebased against the latest develop
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo

### CODE REVIEW

File: src/functions/retroGen.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { ActivitiesService } from "../services/ActivitiesService";
 import { LambdaService } from "../services/LambdaService";
 import { PutObjectCommandOutput } from "@aws-sdk/client-s3";
 import { credentials } from "../handler";
+import {mock} from "sinon";
```
Comment (naathanbrown) [inline src/functions/retroGen.ts]: what is this for?

File: tests/unit/RetroGenFunction.unitTest.ts [hunk 2]
```diff
@@ -64,21 +51,4 @@ describe("Retro Gen Function", () => {
       }
     });
   });
-
```
Comment (naathanbrown) [inline tests/unit/RetroGenFunction.unitTest.ts:67]: why are we removing this?

Comment (Daniel-Searle) [reply tests/unit/RetroGenFunction.unitTest.ts:67]: I removed this as it was testing that process record was throwing an error

File: src/functions/retroGen.ts (hunk 1 above)
Comment (Daniel-Searle) [reply src/functions/retroGen.ts]: un used import now

File: tests/unit/RetroGenFunction.unitTest.ts (hunk 2 above)
Comment (naathanbrown) [reply tests/unit/RetroGenFunction.unitTest.ts:67]: lmao yeah

File: src/functions/retroGen.ts (hunk 1 above)
Comment (naathanbrown) [reply src/functions/retroGen.ts]: but if it's un used why is it here?

Comment (JoblessDan) [reply src/functions/retroGen.ts:13]: Will remove it now

Comment (Daniel-Searle) [reply src/functions/retroGen.ts]: Will remove it now

**Review by naathanbrown — APPROVED** [review]

File: src/functions/retroGen.ts [hunk 3]
```diff
@@ -22,14 +22,15 @@ const retroGen = async (event: any): Promise<void | PutObjectCommandOutput[]> =>
...
   const retroService: RetroGenerationService = new RetroGenerationService(new TestResultsService(new LambdaService(new LambdaClient({ ...credentials }))), new ActivitiesService(new LambdaService(new LambdaClient({ ...credentials }))));
   const retroUploadPromises: Array<Promise<PutObjectCommandOutput>> = [];
   const sharepointAuthenticationService = new SharePointAuthenticationService(rp);
   const sharePointService = new SharePointService(rp);
 
   event.Records.forEach((record: any) => {
-    const recordBody = JSON.parse(JSON.parse(record.body).Message);
-    const visit: any = processRecord(recordBody);
+    const recordBody = JSON.parse(record.body);
+    const visit: any = unmarshall(recordBody?.dynamodb.NewImage);
```
Comment (cb-cs) [inline src/functions/retroGen.ts:33]: Is this filtering now being done in the Eventbridge pipe?

**Review by cb-cs — APPROVED** [review]

**Pull request merged**